src/ql.jl

```diff
@@ -279,6 +292,35 @@ end
 
 (*)(A::QLPackedQ, B::StridedMatrix) = _mul(A, B)
 
+if isdefined(LinearAlgebra, :AdjointQ) # VERSION >= v"1.10-"
+    (*)(A::QLPackedQ, B::StridedVector) = _mul(A, B)
+    function _mul(Q::QLPackedQ, b::AbstractVector)
+        T = promote_type(eltype(Q), eltype(b))
+        if size(Q.factors, 1) == length(b)
+            bnew = LinearAlgebra.copy_similar(b, T)
+        elseif size(Q.factors, 2) == length(b)
+            bnew = [b; zeros(T, size(Q.factors, 1) - length(b))]
+        else
+            throw(DimensionMismatch("vector must have length either $(size(Q.factors, 1)) or $(size(Q.factors, 2))"))
+        end
+        lmul!(convert(AbstractQ{T}, Q), bnew)
+    end
+    # function (*)(A::AbstractMatrix, adjQ::LinearAlgebra.AdjointQ{<:Any,<:QLPackedQ})
+    #     Q = adjQ.Q
+    #     T = promote_type(eltype(A), eltype(adjQ))
+    #     adjQQ = convert(AbstractQ{T}, adjQ)
+    #     if size(A,2) == size(Q.factors, 1)
+    #         AA = LinearAlgebra.copy_similar(A, T)
+    #         return rmul!(AA, adjQQ)
+    #     elseif size(A,2) == size(Q.factors,2)
+    #         return rmul!([A zeros(T, size(A, 1), size(Q.factors, 1) - size(Q.factors, 2))], adjQQ)
+    #     else
+    #         throw(DimensionMismatch("matrix A has dimensions $(size(A)) but Q-matrix B has dimensions $(size(adjQ))"))
+    #     end
+    # end
+    # (*)(u::LinearAlgebra.AdjointAbsVec, Q::LinearAlgebra.AdjointQ{<:Any,<:QLPackedQ}) = (Q'u')'
```

Comment:
There is no `rmul!` now, so I'm not sure if these should be added for completeness or continue to be omitted...